### ISSUE TITLE
Схема бд в postgres

### ISSUE THREAD

**Issue by pipichinka** [dura0ok/Concerts-Suggester#23]
Нам нужно хранить информацию о пользователях: телеграм id, внутренний id, ссылки на плейлисты, список любимых исполнителей, список городов. 
концеры: ссылка на покупку, id исполнителя, дата, город. (возможно рассширение путём добавления нового столбца с какой-нибудь доп информацией)
исполниетли: id, имя, id яндек музыки.(возможно рассширение путём добавления нового столбца айдишников какого-нибудь сервиса).

**Comment by pipichinka** [comment]
Нужно ли добавить в таблицу концертов ссылку на обложку и на карту?.

**Comment by dura0ok** [comment]
CREATE TABLE users (
    user_id SERIAL PRIMARY KEY,
    telegram_id VARCHAR(255) NOT NULL
);

CREATE TABLE track_lists (
    id SERIAL PRIMARY KEY,
    user_id INT REFERENCES users(user_id),
    link VARCHAR(255),
    name VARCHAR(255),
    cover_link VARCHAR(255)
);

CREATE TABLE artists (
    id SERIAL PRIMARY KEY,
    link VARCHAR(255),
    name VARCHAR(255)
);

CREATE TABLE track_lists_artists (
    id SERIAL PRIMARY KEY,
    track_list_id INT REFERENCES track_lists(id),
    artist_id INT REFERENCES artists(id)
);

CREATE TABLE favorite_artists (
    user_id INT REFERENCES users(user_id),
    artist_id INT REFERENCES artists(id),
    PRIMARY KEY (user_id, artist_id)
);

CREATE TABLE cities (
    id SERIAL PRIMARY KEY,
    name VARCHAR(255),
    population INT,
    lat DECIMAL(9, 6),
    lon DECIMAL(9, 6),
    region_name VARCHAR(255),
    region_name_ao VARCHAR(255),
    region_iso_code VARCHAR(255),
    federal_district VARCHAR(255),
    okato VARCHAR(255),
    oktmo VARCHAR(255),
    kladr_id VARCHAR(255),
    fias_id VARCHAR(255),
    place_id VARCHAR(255)
);

CREATE TABLE concerts (
    concert_id SERIAL PRIMARY KEY,
    date TIMESTAMP NOT NULL,
    city_id INT REFERENCES cities(id),
    cost NUMERIC,
    address VARCHAR(255),
    cover_link VARCHAR(255) NOT NULL
);

CREATE TABLE concerts_artists (
    concert_id INT REFERENCES concerts(concert_id),
    artist_id INT REFERENCES artists(id),
    PRIMARY KEY (concert_id, artist_id)
);

Мнение?

**Comment by dura0ok** [comment]
@llirik42 чекни

**Comment by pipichinka** [comment]
Так в табличке с городами остаётся только название и id. 

Из таблички с концертами убираем цену.

Из таблички с артистами убираем link и добавляем ya_music_id.

Из track_lists убираем name и cover_link.

Таблица track_lists_artists переименовывается в таблицу users_artists и в ней будут поля user_id, artist_id(из соответствующих таблиц).

Добавляем таблицу users_cities в которой будут поля user_id и city_id(из соответствующих таблиц).

**Comment by llirik42** [comment]
А что вообще делает `user_id ` в таблице `track_lists`. Типо если один и тот же плейлист нравится нескольким пользователям, то мы будем считать это разными плейлистами?

**Comment by llirik42** [comment]
> Из таблички с концертами убираем цену.

Не согласен.

**Comment by llirik42** [comment]
В таблицу `concerts ` необходимо добавить столбец `afisha_url` (ссылка на покупку билетов). Я бы ещё добавил столбец "название концерта" (потому что концерты могут интересно называться).

**Comment by dura0ok** [comment]
@llirik42 @pipichinka а будет валюта у цены?

**Comment by dura0ok** [comment]
Вопрос к господину @Wooshey1411

**Comment by llirik42** [comment]
> Вопрос к господину @Wooshey1411

Он занят (делает фотошоп), так что не тревожь гОспода

**Comment by pipichinka** [comment]
> > Из таблички с концертами убираем цену.
> 
> Не согласен.

Так, хорошо мы введём цену, это будет средняя цена билетов, самый дорогой билет или самый дешёвый?

**Comment by llirik42** [comment]
> > > Из таблички с концертами убираем цену.
> > 
> > 
> > Не согласен.
> 
> Так, хорошо мы введём цену, это будет средняя цена билетов, самый дорогой билет или самый дешёвый?

Минимальная

**Comment by dura0ok** [comment]
Да, согласен. Минимальная по идее будет кстати, потому что мы будем писать во фронте, что цены от...
--
Отправлено из Mail.ru для Android вторник, 12 марта 2024г., 10:42 +07:00 от Kirill Kondrenko  ***@***.*** :
